src/app/qgsgeometryvalidationdock.cpp

```diff
@@ -182,6 +204,27 @@ void QgsGeometryValidationDock::onCurrentErrorChanged( const QModelIndex &curren
   }
 }
 
+void QgsGeometryValidationDock::onCurrentLayerChanged( QgsMapLayer *layer )
+{
+  // activate icon
+  bool enabled = false;
+  QgsVectorLayer *vl = dynamic_cast<QgsVectorLayer *>( layer );
+  if ( vl && vl->isSpatial() )
+  {
+    const QList<QgsGeometryCheckFactory *> topologyCheckFactories = QgsAnalysis::instance()->geometryCheckRegistry()->geometryCheckFactories( vl, QgsGeometryCheck::LayerCheck, QgsGeometryCheck::Flag::AvailableInValidation );
+    const QStringList activeChecks = vl->geometryOptions()->geometryChecks();
+    for ( const QgsGeometryCheckFactory *factory : topologyCheckFactories )
```

Comment:
Let's send this through the geometry validation service which already keeps track of that.